### PR TITLE
Dc 7 db

### PR DESCRIPTION
**Pull request by umamiMike**
updates for using direnv  and tmux for dev env

- adds .envrc to allow automatic shell  with nix using zsh
https://github.com/direnv/direnv

assuming tmux is not installed
 and nix is [nix-package-manager](https://nixos.org/) is installed 
`bash <(curl -L https://nixos.org/nix/install)`

1. enter the root directory
2. type `tmux`
 should show `command not found`
3. type `nix-shell` 
should be able to use tmux
4. run  `./startup-processes`
should read something like `db-compare-dev: 3 windows ... `

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,9 @@ which I think is neat.
 
 It is very much a work in progress at the moment
 
+1. Install nix ` curl -L https://nixos.org/nix/install | sh `
+2. cd into the project directory
+3. Run `nix-shell`
```
Comment (psbanka) [inline README.md:36]: This does not work:
```
❯ nix-shell
error: undefined variable 'go_1_13' at /Users/peba/play/db-compare/default.nix:9:8
```

File: db-compare-server/api.go [hunk 2]
```diff
@@ -4,94 +4,80 @@ import (
...
 	}
 
-	jsonData, err := json.MarshalIndent(response, "", "  ")
-	if err != nil {
-		fmt.Println(err)
-	}
-	setJsonApiHeaders(w)
-	w.Write(jsonData)
+	// storage.Set(ID, response)
+	// setJsonApiHeaders(w)
```
Comment (psbanka) [inline db-compare-server/api.go]: let's not leave commented code in a commit. delete it! if you need to see this later, you can always use git to look at old code.

File: db-compare-server/db/db_test.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"fmt"
+	"testing"
+)
+
+func TestNewBadger(t *testing.T) {
+	storageDir := "/tmp/badger"
+	badger := NewBadger(storageDir)
+	fmt.Printf("%#v", badger)
+	//assert.Equal(t, storageDir, badger.opts.Dir, "new badger should return Badger with values, opts containaing storageDir")
```
Comment (psbanka) [inline db-compare-server/db/db_test.go]: why is this assert commented out?

File: db-compare-server/dev-build.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+  $(killall db-compare-server)
+clear
+  go build && echo " \e[1m\e[36mapp rebuilt successfully\e[0m" || echo -e "\n\n\e[1m[91mrebuild failed\n\n"
+  ./db-compare-server & 
+inotifywait ./ -q -m -e close_write | while read events; do
```
Comment (psbanka) [inline db-compare-server/dev-build.sh]: `inotifywait` is not installed by default. Need instructions for installation.

Comment (psbanka) [reply db-compare-server/dev-build.sh]: `fswait` might be a better cross-platform alternative https://stackoverflow.com/questions/1515730/is-there-a-command-like-watch-or-inotifywait-on-the-mac

File: db-compare-server/main.go [hunk 5]
```diff
@@ -41,10 +28,9 @@ func main() {
...
 		WriteTimeout: 10 * time.Second,
 	}
-	fmt.Printf("server  started at http://localhost%v/\n ", server.Addr)
-
-	r.Post("/datasources", datasourcesHandler)
-	r.Post("/queries", queriesPostHandler)
+	fmt.Printf("server started at http://localhost%v/\n ", server.Addr)
+	r.Get("/", indexHandler)
+	r.Post("/datasources", datasourcesPostHandler)
+	// r.Post("/queries", queriesPostHandler)
```
Comment (psbanka) [inline db-compare-server/main.go]: remove commented code

File: logfile [hunk 6]
```diff
@@ -1,3 +0,0 @@
-2020-07-17 17:00:48.848 PDT [8761] LOG:  listening on IPv4 address "127.0.0.1", port 5432
-2020-07-17 17:00:48.852 PDT [8761] FATAL:  could not create lock file "/run/postgresql/.s.PGSQL.5432.lock": No such file or directory
-2020-07-17 17:00:48.852 PDT [8761] LOG:  database system is shut down
```
Comment (psbanka) [inline logfile:3]: does this file need to be checked in?

**Review by psbanka — CHANGES_REQUESTED** [review]
Pretty good. Some of the instructions are incomplete and do not work on other systems. Some commented code that should be removed.

File: README.md (hunk 1 above)
Comment (umamiMike) [reply README.md:36]: should be fixed now.

File: db-compare-server/dev-build.sh (hunk 4 above)
Comment (umamiMike) [reply db-compare-server/dev-build.sh]: updated to use fswatch....an interesting little piece of work.

File: logfile (hunk 6 above)
Comment (umamiMike) [reply logfile:3]: removed

**Pull request merged**